server/fcm/fcm.go

```diff
@@ -43,408 +25,110 @@ type Config struct {
 	APIKey               *string
 	Workers              *int
 	Endpoint             *string
+	Prefix               *string
 	AfterMessageDelivery protocol.MessageDeliveryCallback
 }
 
 // Connector is the structure for handling the communication with Firebase Cloud Messaging
-type Connector struct {
-	Sender               gcm.Sender
-	AfterMessageDelivery protocol.MessageDeliveryCallback
-	router               router.Router
-	cluster              *cluster.Cluster
-	kvStore              kvstore.KVStore
-	prefix               string
-	pipelineC            chan *subscriptionMessage
-	stopC                chan bool
-	nWorkers             int
-	wg                   sync.WaitGroup
-	subscriptions        map[string]*subscription
-}
-
-// New creates a new *Connector without starting it
-func New(router router.Router, prefix string, config Config) (*Connector, error) {
-	logger.WithField("count", *config.Workers).Debug("FCM workers")
-	kvStore, err := router.KVStore()
+type fcm struct {
+	Config
+	connector.Connector
+	// sender connector.Sender
+}
+
+// New creates a new *fcm and returns it as an connector.ReactiveConnector
+func New(router router.Router, sender connector.Sender, config Config) (connector.ReactiveConnector, error) {
+	baseConn, err := connector.NewConnector(router, sender, connector.Config{
+		Name:       "fcm",
+		Schema:     schema,
+		Prefix:     *config.Prefix,
+		URLPattern: fmt.Sprintf("/{device_token}/{user_id}/{%s:.*}", connector.TopicParam),
```

Comment:
please extract `device_token` and `user_id` as constants
reuse them in this package (see for instance line 129, below, maybe some other places)